tests/helpers/utils.py

```diff
@@ -114,15 +114,22 @@ def inner_f(queue, **kwargs):
 
 
 @contextmanager
-def no_warning_call(warning_type, match: Optional[str] = None):
+def no_warning_call(expected_warning: Type[Warning] = UserWarning, match: Optional[str] = None):
     with pytest.warns(None) as record:
         yield
 
+    if match is None:
         try:
-            w = record.pop(warning_type)
-            if not (match and match in str(w.message)):
-                return
+            w = record.pop(expected_warning)
         except AssertionError:
             # no warning raised
             return
-        raise AssertionError(f"`{warning_type}` was raised: {w}")
+    else:
+        for w in record.list:
+            if w.category is expected_warning and match in w.message.args[0]:
+                break
+        else:
+            return
```

Comment:
this is a bit taff to follow :rabbit: not everyone know meaning `else` for `for` loop

Reply:
if they don't, should search and mark it as TIL :smile: